app/src/main/java/app/revanced/manager/ui/component/InstallerDialog.kt

```diff
@@ -0,0 +1,144 @@
+package app.revanced.manager.ui.component
+
+import android.content.pm.PackageInstaller
+import androidx.annotation.RequiresApi
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Check
+import androidx.compose.material.icons.outlined.ErrorOutline
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import app.revanced.manager.R
+import app.revanced.manager.ui.viewmodel.InstallerViewModel
+import com.github.materiiapps.enumutil.FromValue
+
+private typealias InstallerDialogButtonHandler = ((vm: InstallerViewModel) -> Unit)
+private typealias InstallerDialogButton = @Composable (vm: InstallerViewModel) -> Unit
+
+
+@Composable
+fun InstallerDialog(
+    vm: InstallerViewModel,
+) {
+    val dialogKind = DialogKind.fromValue(vm.installerStatus!!) ?: DialogKind.FAILURE
+
+    AlertDialog(
+        onDismissRequest = {},
+        confirmButton = {
+            dialogKind.confirmButton(vm)
+        },
+        dismissButton = {
+            dialogKind.dismissButton?.invoke(vm)
+        },
+        icon = {
+            Icon(dialogKind.icon, null)
+        },
+        title = {
+            Text(
+                text = stringResource(dialogKind.title),
+                style = MaterialTheme.typography.headlineSmall.copy(textAlign = TextAlign.Center),
+                color = MaterialTheme.colorScheme.onSurface,
+            )
+        },
+        text = {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.spacedBy(16.dp),
+                content = {
+                    Text(stringResource(dialogKind.contentStringResId))
+                }
+            )
+        }
+    )
+}
+
+private fun installerDialogButton(
+    @StringRes buttonStringResId: Int,
+    handler: InstallerDialogButtonHandler = { },
+): InstallerDialogButton = { vm ->
+    TextButton(onClick = {
+        vm.showInstallerDialog = false
+        handler(vm)
+    }) { Text(stringResource(buttonStringResId)) }
+}
```

Comment:
How about puttiing the `handler` in the enum instead of a composable?

Reply:
If we put the handler in the enum, then we would need an additional field for `buttonStringResId`, whereas if we keep `InstallerDialogButton` in the enum, you can pass `buttonStringResId` and `handler`.